### DISCUSSION TITLE
What is the benefit to pass into a setter a modifier function ?

### DISCUSSION THREAD

**Discussion by yw662** [solidjs/solid#993] [Q&A]
like,
```ts
const [count, setCount] = createSignal(0)
...
const modifier = (count: number) => count + 1
...
setCount(modifier(count()))
...
```
vs
```ts
const [count, setCount] = createSignal(0)
...
setCount(modifier)
...
```

What is the benefit to support the later form ?

**Comment by yw662** [comment]
One of a benefit of the former one (or, dropping support for the later one) is related to  #980, so you won't `Exclude<T, Function>`

**Comment by ryansolid** [comment]
The read of the current value is `untrack`'d which can prevent unintended reads inside computations. This is even more pronounced when looking at stores which are deep, and keeping the API consistent.
